spektral/layers/convolutional/censnet_conv.py

```diff
@@ -0,0 +1,259 @@
+import tensorflow as tf
+
+from spektral.layers import ops
+from spektral.layers.convolutional.conv import Conv
+from spektral.utils.convolution import gcn_filter, incidence_matrix, line_graph
+
+
+class CensNetConv(Conv):
+    r"""
+    Convolution layer from [CensNet](
+    https://ieeexplore.ieee.org/stamp/stamp.jsp?tp=&arnumber=9224195)
+
+    This implements both the node and edge propagation rules as a single layer.
+
+    **Input**
+
+    - Node features of shape `([batch], n_nodes, n_node_features)`;
+    - A tuple containing:
+        - Modified Laplacian of shape `([batch], n_nodes, n_nodes)`; can be
+            computed with `spektral.utils.convolution.gcn_filter`.
+        - Modified line graph Laplacian of shape `([batch], n_edges, n_edges)`;
+            can be computed with `spektral.utils.convolution.line_graph` and
+            `spektral.utils.convolution.gcn_filter`.
+        - Incidence matrix of shape `([batch], n_nodes, n_edges)`; can be
+            computed with `spektral.utils.convolution.incidence_matrix`.
+    - Edge features of shape `([batch], n_edges, n_edge_features)`;
```

Comment:
I have a question about this. I've noticed that the one other layer that supports edge features in batch mode wants them in "sparse" form, i.e. with a shape of `([batch], n_nodes, n_nodes, n_edge_features)` instead. I could change this layer to expect them in that mode too, but I'd just end up immediately converting them back in order to perform the layer operation. Is it okay to expect them in this form, even though it makes things slightly inconsistent?